### PR TITLE
Replace rounds.validateBlockSlot with dpos.verifyBlockForger - Closes #4155, #4156

### LINKED ISSUE #4155: Replace rounds.validateBlockSlot by dpos.verifyBlockForger
### Description

Replace `rounds.validateBlockSlot` by `dpos.verifyBlockForger` (to be implemented by https://github.com/LiskHQ/lisk-sdk/issues/4144)

### Motivation

Rounds module is being replaced by Dpos module as part of BFT implementation

### Acceptance Criteria

- Every use of `rounds.validateBlockSlot` should be replaced by `dpos.verifyBlockForger`
- Tests should be updated to use `dpos.verifyBlockForger` instead of `rounds.validateBlockSlot`
- Should sync on testnet at least up to height 10000

### Additional Information
--

### PR DESCRIPTION
**Pull request by michielmulders**
### What was the problem?
We need to replace the old functions with the new DPoS variants.

### How did I solve it?
Replacing rounds logic with DPoS logic.

### Review checklist

- [x] The PR resolves #4155, #4156 
- [x] All new code is covered with unit tests
- [x] Relevant integration / functional tests are added
- [x] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [x] Documentation has been added/updated

### CODE REVIEW

**Comment by michielmulders** [comment]
This PR can be reviewed but is on hold til #4144 gets merged. Danke! :)

File: framework/test/mocha/unit/modules/chain/blocks/verify.js [hunk 1]
```diff
@@ -1140,36 +1146,36 @@ describe('blocks/verify', () => {
...
-	describe('validateBlockSlot', () => {
+	describe('verifyBlockForger', () => {
 		const dummyBlock = { id: 1 };
 
-		describe('when rounds.validateBlockSlot fails', () => {
+		describe('when dpos.verifyBlockForger fails', () => {
 			beforeEach(async () => {
-				roundsModuleStub.validateBlockSlot.rejects(
+				dposModuleStub.verifyBlockForger.rejects(
 					new Error('validateBlockSlot-ERR'),
```
Comment (lsilvs) [inline framework/test/mocha/unit/modules/chain/blocks/verify.js]:
```suggestion
					new Error('verifyBlockForger-ERR'),
```

File: framework/test/mocha/unit/modules/chain/blocks/verify.js [hunk 2]
```diff
@@ -1140,36 +1146,36 @@ describe('blocks/verify', () => {
...
 
 			it('should call a callback with error', async () => {
 				try {
-					await blocksVerify.validateBlockSlot(dummyBlock);
+					await blocksVerify.verifyBlockForger(dummyBlock);
 				} catch (error) {
 					expect(roundsModuleStub.fork).calledWith(dummyBlock, 3);
-					expect(roundsModuleStub.validateBlockSlot).calledWith(dummyBlock);
+					expect(dposModuleStub.verifyBlockForger).calledWith(dummyBlock);
 					expect(error.message).to.equal('validateBlockSlot-ERR');
```
Comment (lsilvs) [inline framework/test/mocha/unit/modules/chain/blocks/verify.js]:
```suggestion
					expect(error.message).to.equal('verifyBlockForger-ERR');
```

File: framework/src/modules/chain/rounds/delegates.js [hunk 3]
```diff
@@ -145,50 +121,6 @@ class Delegates {
...
-	 * Generates delegate list and checks if block generator public key matches delegate id.
-	 *
-	 * @param {block} block
-	 * @param {function} cb - Callback function
-	 * @returns {setImmediateCallback} cb, err
-	 * @todo Add description for the params
-	 */
-	async validateBlockSlot(block) {
-		const round = this.slots.calcRound(block.height);
-		const activeDelegates = await this.generateDelegateList(
```
Comment (lsilvs) [inline framework/src/modules/chain/rounds/delegates.js]: As `generateDelegateList` is not used anymore, all the following functions in this file can be removed:
```
generateDelegateList
updateDelegateListCache
clearDelegateListCache
getKeysSortByVote
getDelegatesFromPreviousRound
```

Comment (michielmulders) [reply framework/src/modules/chain/rounds/delegates.js]: Decided to remove after `undo` and `apply` are implemented as the `getOutsiders` function is still using it.

Comment (michielmulders) [reply framework/src/modules/chain/rounds/delegates.js]: Created issue for this and added to Yatki's parent issue: https://github.com/LiskHQ/lisk-sdk/issues/4168

Comment (lsilvs) [reply framework/src/modules/chain/rounds/delegates.js]: There is an issue for this already: https://github.com/LiskHQ/lisk-sdk/issues/3685

**Pull request merged**